pymongo/bulk.py

```diff
@@ -149,31 +181,31 @@ def __init__(self, collection, ordered, bypass_document_validation, comment=None
         self.let = let
         if self.let is not None:
             common.validate_is_document_type("let", self.let)
-        self.comment = comment
-        self.ordered = ordered
-        self.ops = []
-        self.executed = False
-        self.bypass_doc_val = bypass_document_validation
-        self.uses_collation = False
-        self.uses_array_filters = False
-        self.uses_hint_update = False
-        self.uses_hint_delete = False
-        self.is_retryable = True
-        self.retrying = False
-        self.started_retryable_write = False
+        self.comment: Optional[str] = comment
+        self.ordered: bool = ordered
+        self.ops: List[Tuple[int, Union[MutableMapping[str, Any], RawBSONDocument]]] = []
+        self.executed: bool = False
+        self.bypass_doc_val: bool = bypass_document_validation
+        self.uses_collation: bool = False
+        self.uses_array_filters: bool = False
+        self.uses_hint_update: bool = False
+        self.uses_hint_delete: bool = False
+        self.is_retryable: bool = True
+        self.retrying: bool = False
+        self.started_retryable_write: bool = False
         # Extra state so that we know where to pick up on a retry attempt.
         self.current_run = None
         self.next_run = None
 
     @property
-    def bulk_ctx_class(self):
+    def bulk_ctx_class(self) -> Type[_BulkWriteContext]:
```

Comment:
Should this be just `_BulkWriteContext` instead since we're using future annotations?

Reply:
No, it needs to be the class, not an instance

Reply:
I don't think so. The function is returning the class itself..not an instance of the class. If it were an instance of the class then I think it would be `_BulkWriteContext` but since the return statement is like `return _BulkWriteContext`, I think the return type has to be `Type[_BulkWriteContext]`